skyportal/handlers/api/earthquake.py

```diff
@@ -0,0 +1,616 @@
+    else:
+        if not event.is_accessible_by(user, mode="update"):
+            raise ValueError(
+                "Insufficient permissions: Earthquake event can only be updated by original poster"
+            )
+
+    earthquake_notice = EarthquakeNotice(
+        event_id=event_id,
+        lat=latitude,
+        lon=longitude,
+        depth=depth,
+        magnitude=magnitude,
+        date=date,
+        sent_by_id=user.id,
+    )
+    session.add(earthquake_notice)
+    session.commit()
+
+    return event.id
+
+
+class EarthquakeStatusHandler(BaseHandler):
+    @auth_or_token
+    def get(self):
+        """
+        ---
+        description: Get all Earthquake status tags
+        tags:
+          - earthquakeevents
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        with self.Session() as session:
+            statuses = (
+                session.scalars(sa.select(EarthquakeEvent.status).distinct())
+                .unique()
+                .all()
+            )
+            return self.success(data=statuses)
+
+
+class EarthquakeHandler(BaseHandler):
+    @auth_or_token
+    def post(self):
+        """
+        ---
+        description: Ingest EarthquakeEvent
+        tags:
+          - earthquakeevents
+          - earthquakenotices
+        requestBody:
+          content:
+            application/json:
+              schema: EarthquakeEventNoID
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+                properties:
+                  data:
+                    type: object
+                    properties:
+                      gcnevent_id:
+                        type: integer
+                        description: New Earthquake ID
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+        data = self.get_json()
+        if 'xml' not in data:
+            required_keys = {
+                'date',
+                'event_id',
+                'latitude',
+                'longitude',
+                'depth',
+                'magnitude',
+            }
+            if not required_keys.issubset(set(data.keys())):
+                return self.error(
+                    "Either xml or (event_id, latitude, longitude, depth, magnitude) must be present in data to parse EarthquakeEvent"
+                )
+
+        with self.Session() as session:
+            if 'xml' in data:
+                event_id = post_earthquake_from_xml(
+                    data['xml'], self.associated_user_object.id, session
+                )
+            else:
+                event_id = post_earthquake_from_dictionary(
+                    data, self.associated_user_object.id, session
```

Comment:
If you only ever post using `self.Session` you can use the associated user on that session instead of passing it in.

Reply:
yeah will also be used in the service, so also need a DBSession() style.